aten/src/ATen/version_update_mode.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include <c10/macros/Macros.h>
+
+namespace at {
+
+struct CAFFE2_API VersionUpdateMode {
```

Comment:
This needs a comment

Reply:
There are some places in code where it is not sound to toggle the enabled setting. We should explain when this is the case. It's something like, you're not allowed to set this when you're "inside" a Variable operation.